spring-cloud-skipper-client/src/main/java/org/springframework/cloud/skipper/client/SkipperClientConfiguration.java

```diff
@@ -0,0 +1,58 @@
+/*
+ * Copyright 2017 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.springframework.cloud.skipper.client;
+
+import org.springframework.boot.context.properties.EnableConfigurationProperties;
+import org.springframework.boot.web.client.RestTemplateBuilder;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.hateoas.config.EnableHypermediaSupport;
+import org.springframework.http.converter.HttpMessageConverter;
+import org.springframework.http.converter.json.MappingJackson2HttpMessageConverter;
+import org.springframework.web.client.RestTemplate;
+
+/**
+ * @author Mark Pollack
+ */
+@Configuration
+@EnableConfigurationProperties(SkipperClientConfigurationProperties.class)
+@EnableHypermediaSupport(type = EnableHypermediaSupport.HypermediaType.HAL)
+public class SkipperClientConfiguration {
+
+	@Bean
+	public RestTemplate restTemplate(RestTemplateBuilder restTemplateBuilder) {
+		RestTemplate restTemplate = restTemplateBuilder.build();
+		boolean containsMappingJackson2HttpMessageConverter = false;
+
+		for (HttpMessageConverter<?> converter : restTemplate.getMessageConverters()) {
+			if (converter instanceof MappingJackson2HttpMessageConverter) {
+				containsMappingJackson2HttpMessageConverter = true;
+			}
+		}
+
+		if (!containsMappingJackson2HttpMessageConverter) {
+			throw new IllegalArgumentException(
+					"The RestTemplate does not contain a required " + "MappingJackson2HttpMessageConverter.");
+		}
+		return restTemplate;
+	}
+
+	@Bean
+	public SkipperClient skipperClient(SkipperClientConfigurationProperties properties) {
+		return new DefaultSkipperClient(properties.getHome());
```

Comment:
I guess we aren't even using the `DefaultSkipperClient` before we end up creating a new one anyway on ConfigCommands that is run on startup.  Will fix though.